### PR TITLE
[CI] error reference publisher for meshkit

### PR DESCRIPTION
**Pull request by DelusionalOptimist**
Signed-off-by: Rudraksh Pareek <rudrakshpareek3601@gmail.com>

**Description**
Integrating the error codes utility into meshkit's CI.
Also see https://github.com/layer5io/meshery/pull/3296
This PR fixes #

**Notes for Reviewers**
* Runs error codes utility on every commit to the master branch.
* Updates the files generated by it in this repo.
* Updates meshery docs and pushes them to meshery repo.

*Commit e8a3f5f is just for demo*

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: .github/workflows/error-ref-publisher.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      - uses: actions/checkout@v2
+
+      - name: Setup Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: ${{ secrets.GO_VERSION }}
+
+      - name: Run utility
+        run: |
+          go run github.com/layer5io/meshkit/cmd/errorutil -d . analyze --skip-dirs meshery
```
Comment (kumarabd) [inline .github/workflows/error-ref-publisher.yaml:21]: You could update the `make` command and use them here @DelusionalOptimist

File: .github/workflows/error-ref-publisher.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      - name: Commit changes
+        uses: stefanzweifel/git-auto-commit-action@v4
+        with:
+          commit_user_name: l5io
+          commit_user_email: ci@layer5.io
+          commit_author: ${{ github.actor }} <${{ github.actor }}@users.noreply.github.com>
+          commit_options: '--signoff'
+          commit_message: "run error codes utility"
+
+      - name: Checkout meshery
```
Comment (kumarabd) [inline .github/workflows/error-ref-publisher.yaml:33]: Does it checkout `master` branch by default? We would want to make sure that happens

File: .github/workflows/error-ref-publisher.yaml (hunk 1 above)
Comment (DelusionalOptimist) [reply .github/workflows/error-ref-publisher.yaml]: Done :+1:

File: .github/workflows/error-ref-publisher.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        uses: actions/checkout@v2
+        with:
+          repository: 'layer5io/meshery'
```
Comment (leecalcote) [inline .github/workflows/error-ref-publisher.yaml:36]:
```suggestion
          repository: 'meshery/meshery'
```

File: .github/workflows/error-ref-publisher.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+          # token with write access to meshery repository
+          token: ${{ secrets.PAT }}
```
Comment (leecalcote) [inline .github/workflows/error-ref-publisher.yaml:38]:
```suggestion
          token: ${{ secrets.GH_ACCESS_TOKEN }}
```

File: .github/workflows/error-ref-publisher.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+      # to push changes to meshery docs
+      - name: Update and push docs
+        run: |
+          echo '{ "errors_export": '$(cat errorutil_errors_export.json)' }' | jq > ./meshery/docs/_data/errorref/meshkit_errors_export.json
+
+          cd ./meshery
+          git config user.name l5io
+          git config user.email ci@layer5.io
+          git add ./docs/_data/errorref/meshkit_errors_export.json
+          git commit -m "[Docs] update error reference for Meshkit library" --signoff
```
Comment (leecalcote) [inline .github/workflows/error-ref-publisher.yaml:51]:
```suggestion
          git commit -m "[Docs] update error reference for MeshKit library" --signoff
```

**Review by leecalcote — APPROVED** [review]

File: .github/workflows/error-ref-publisher.yaml [hunk 6]
```diff
@@ -0,0 +1,53 @@
+name: Error Reference Publisher
```
Comment (leecalcote) [inline .github/workflows/error-ref-publisher.yaml:1]:
```suggestion
name: Error Code Reference Publisher
```

File: .github/workflows/error-ref-publisher.yaml [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+      # to push changes to meshery docs
+      - name: Update and push docs
+        run: |
+          echo '{ "errors_export": '$(cat errorutil_errors_export.json)' }' | jq > ./meshery/docs/_data/errorref/meshkit_errors_export.json
+
+          cd ./meshery
+          git config user.name l5io
+          git config user.email ci@layer5.io
+          git add ./docs/_data/errorref/meshkit_errors_export.json
+          git commit -m "[Docs] update error reference for MeshKit library" --signoff
```
Comment (leecalcote) [inline .github/workflows/error-ref-publisher.yaml:51]:
```suggestion
          git commit -m "[Docs] Error Code Reference: Updated codes for MeshKit library" --signoff
```

**Review by leecalcote — APPROVED** [review]

**Comment by leecalcote** [comment]
Uh-oh, a broken link in the welcome.yml. @debjani, perhaps, you would like to fix?

**Pull request merged**